app/services/schools/advice/baseload_service.rb

```diff
@@ -0,0 +1,191 @@
+
+      def estimated_savings(versus: :benchmark_school)
+        benchmark_service.estimated_savings(versus: versus)
+      end
+
+      #Calculate the annual average baseload for every year
+      def annual_average_baseloads
+        start_date, end_date = date_range
+        (start_date.year..end_date.year).map do |year|
+          end_of_year = Date.new(year).end_of_year
+          baseload_service = Baseload::BaseloadCalculationService.new(aggregate_meter, end_of_year)
+          {
+            year: year,
+            baseload: baseload_service.average_baseload_kw(period: :year),
+            baseload_usage: baseload_service.annual_baseload_usage
+          }
+        end
+      end
+
+      def baseload_meter_breakdown
+        meter_breakdown_service = Baseload::BaseloadMeterBreakdownService.new(@meter_collection)
+        baseloads = meter_breakdown_service.calculate_breakdown
+        meter_breakdowns = {}
+        baseloads.meters.each do |mpan_mprn|
+          baseload_service = Baseload::BaseloadCalculationService.new(@meter_collection.meter?(mpan_mprn), end_of_previous_year)
+          previous_year_baseload = baseload_service.average_baseload_kw(period: :year)
+          meter_breakdowns[mpan_mprn] = build_meter_breakdown(mpan_mprn, baseloads, previous_year_baseload)
+        end
+        meter_breakdowns
+      end
+
+      #helper for building "all meters" / total row for meter breakdown table
+      def meter_breakdown_table_total
+        baseload_usage = annual_baseload_usage
+        previous_year_baseload = previous_year_average_baseload_kw
+        baseload_kw = average_baseload_kw
+        OpenStruct.new(
+          baseload_kw: baseload_kw,
+          baseload_cost_£: baseload_usage.£,
+          percentage_baseload: 1.0,
+          baseload_previous_year_kw: previous_year_baseload,
+          baseload_change_kw: baseload_kw - previous_year_baseload
+        )
+      end
+
+      def seasonal_variation
+        calculate_seasonal_variation(aggregate_meter, asof_date)
+      end
+
+      def seasonal_variation_by_meter
+        variation_by_meter = {}
+        if @meter_collection.electricity_meters.count > 1
+          @meter_collection.electricity_meters.each do |meter|
+            variation_by_meter[meter.mpan_mprn] = calculate_seasonal_variation(meter, meter.amr_data.end_date, true)
+          end
+        end
+        variation_by_meter
+      end
+
+      def intraweek_variation
+        calculate_intraweek_variation(aggregate_meter, asof_date)
+      end
+
+      def intraweek_variation_by_meter
+        variation_by_meter = {}
+        if @meter_collection.electricity_meters.count > 1
+          @meter_collection.electricity_meters.each do |meter|
+            variation_by_meter[meter.mpan_mprn] = calculate_intraweek_variation(meter, meter.amr_data.end_date, true)
+          end
+        end
+        variation_by_meter
+      end
+
+      private
+
+      def asof_date
+        @asof_date ||= date_range[1]
```

Comment:
I've left a comment above on a possible use of openstruct for date_range (e.g. `date_range.end_date`)